src/main/java/org/jitsi/videobridge/shim/ConferenceShim.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright @ 2019-Present 8x8, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge.shim;
+
+import net.java.sip.communicator.impl.protocol.jabber.extensions.colibri.*;
+import org.jitsi.service.neomedia.*;
+import org.jitsi.videobridge.*;
+
+import java.util.*;
+
+/**
+ * Handles Colibri-related logic for a {@link Conference}, e.g.
+ * creates/expires contents, describes the conference in XML.
+ *
+ * @author Brian Baldino
+ * @author Boris Grozev
+ */
+public class ConferenceShim
+{
+    public final Conference conference;
+
+    private final Map<MediaType, ContentShim> contents = new HashMap<>();
+    private final Map<String, ChannelBundleShim> channelBundles = new HashMap<>();
+
+    public ConferenceShim(Conference conference)
+    {
+        this.conference = conference;
+        System.out.println("ConferenceShim created conference " + conference.getID());
+    }
+
+    public ContentShim getOrCreateContent(MediaType type) {
+        synchronized (contents)
+        {
+            return contents.computeIfAbsent(type,
+                    key -> new ContentShim(getConference(), type));
+        }
+    }
+    public Conference getConference()
+    {
+        return conference;
+    }
+
+    public Collection<ContentShim> getContents()
+    {
+        synchronized (contents)
+        {
+            return new ArrayList<>(contents.values());
+        }
+    }
+
+    public AbstractEndpoint getOrCreateEndpoint(String endpointId)
```

Comment:
i thought the idea with these changes was to change the shims from being the primary part (e.g. responsible for creating their 'real life' counterparts conference, endpoints, etc.) to being more 'ornamental' (just used for the signaling) and therefore we'd enforce all creation happened through conference, etc. and the shim was just updated when it needed to be (or would query the real thing when it needed to).  how do you see the responsibilities for the shim vs. the real classes at this point?

Reply:
sorry, i wrote this before checking out the other commit.  after seeing the rest, i think that's what you did there...am i understanding it right? things like content and channel (which don't exist in the new scheme) will be created by the shims, but other things (conference, endpoint) will be created directly via the real thing and they will create their own shims? is that right?

Reply:
Yes, exactly